.github/actions/cross-branch-migration-test/action.yml

```diff
@@ -0,0 +1,43 @@
+name: Cross-branch migration test
+
+runs:
+  using: "composite"
+  steps:
+    - name: Check out master request branch
+      uses: actions/checkout@v4
+      with:
+        ref: master
+
+    - name: Prepare backend environment
+      uses: ./.github/actions/prepare-backend
+      with:
+        m2-cache-key: cross-branch-migrations
+
+    - name: Run backend on master
+      run: clojure -M:run:ee & echo $! > metabase_pid
+      shell: bash
+
+    - name: Wait for Metabase to start
+      run: while ! curl 'http://localhost:3000/api/health' | grep '{"status":"ok"}'; do sleep 1; done
+      shell: bash
+
+    - name: Stop Metabase on master
+      run: kill $(cat metabase_pid)
+      shell: bash
+
+    - name: Check out pull request branch
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ github.head_ref }}
+
+    - name: Run backend on pull request branch
+      run: clojure -M:run:ee & echo $! > metabase_pid
+      shell: bash
+
+    - name: Wait for Metabase to start
+      run: while ! curl 'http://localhost:3000/api/health' | grep '{"status":"ok"}'; do sleep 1; done
```

Comment:
I think you can export the PID from the last step, and then you can check if it is still running in this loop to be able to exit the action early if it already exited.

see https://stackoverflow.com/questions/36882936/how-to-exit-the-bash-script-when-one-of-its-background-processes-has-error

Reply:
cool — we have this pattern of polling the health check endpoint with a timeout in a few other places, so if this works I can change those spots too